cpp/src/io/orc/orc.h

```diff
@@ -131,6 +131,67 @@ struct Metadata {
   std::vector<StripeStatistics> stripeStats;
 };
 
+int static constexpr encode_field_number(int field_number, uint8_t field_type) noexcept
```

Comment:
Yeah I feel like field number should be an enum.  Looks like field_type already is an enum, but an anonymous one.

Reply:
Yeah, field type can be passed as enum, I'll look into that.